src/Microsoft.AspNetCore.SignalR.Common/Internal/Protocol/Utf8BufferTextReader.cs

```diff
@@ -11,47 +11,82 @@ namespace Microsoft.AspNetCore.SignalR.Internal.Protocol
     internal class Utf8BufferTextReader : TextReader
     {
         private ReadOnlyMemory<byte> _utf8Buffer;
+        private Decoder _decoder;
 
-        public Utf8BufferTextReader(ReadOnlyMemory<byte> utf8Buffer)
+        [ThreadStatic]
+        private static Utf8BufferTextReader _cachedInstance;
+
+#if DEBUG
+        private bool _inUse;
```

Comment:
Isn't this going to block up if two threads are running parses? This should be thread static too, right?

Reply:
It's a field on the class, and the class is on the thread static.

Reply:
![https://media1.tenor.com/images/ed14f0a8a9528037ecde16778bc5df54/tenor.gif?itemid=6233670](https://media1.tenor.com/images/ed14f0a8a9528037ecde16778bc5df54/tenor.gif?itemid=6233670)